app/controllers/users_controller.rb

```diff
@@ -0,0 +1,52 @@
+class UsersController < ApplicationController
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'users/show'
+  end
+
+ get '/signup' do
+   if !logged_in?
+     erb :'users/create_user', locals: {message: "Please sign up before you sign in"}
+   else
+     redirect to '/tweets'
+   end
+ end
+
+ post '/signup' do
+   if params[:username] == "" || params[:email] == "" || params[:password] == ""
+     redirect to '/signup'
+   else
+     @user = User.create(:username => params[:username], :email => params[:email], :password => params[:password])
+     session[:user_id] = @user.id
+     redirect to '/tweets'
+   end
+ end
+
+ get '/login' do
+   if !logged_in?
+     erb :'users/login'
+   else
+     redirect '/tweets'
+   end
+ end
+
+ post '/login' do
+   user = User.find_by(:username => params[:username])
```

Comment:
what if two users have the same username would this still work correctly? Look into Active Record validations for uniqueness